content/_code-samples/airgapped-wallet/py/airgapped-wallet.py

```diff
@@ -0,0 +1,203 @@
+import os
+import base64
+import qrcode
+import platform
+from PIL import Image
+from pathlib import Path, PureWindowsPath, PurePath
+from cryptography.fernet import Fernet
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.kdf.pbkdf2 import PBKDF2HMAC
+from xrpl import wallet
+from xrpl.core import keypairs
+from xrpl.utils import xrp_to_drops
+from xrpl.models.transactions import Payment
+from xrpl.transaction import safe_sign_transaction
+
+
+def create_wallet():
+    """
+    Generates a keypair
+    """
+
+    seed = keypairs.generate_seed()
+    pub, priv = keypairs.derive_keypair(seed)
+
+    address = keypairs.derive_classic_address(pub)
+    private = keypairs.derive_classic_address(priv)
+    print(
+        f"\n\n       XRP WALLET CREDENTIALS"
+        f"\n  Wallet Address: {address}"
+        f"\n     Private Key: {private}"
+        f"\n            Seed: {seed}"
+          )
+
+    return address, private, seed
+
+
+def sign_transaction(_xrp_amount, _destination, _ledger_seq, _wallet_seq, password):
+    """
+    Signs transaction and returns signed transaction blob in QR code
+    """
+
+    with open(get_path("/Wallet/private.txt"), "r") as f:
+        _seed = f.read()
+        _seed = bytes.fromhex(_seed)
+
+    with open(get_path("/Wallet/salt.txt"), "rb") as f:
+        salt = f.read()
+
+    # Line 49-58: initialize key
+    kdf = PBKDF2HMAC(
+        algorithm=hashes.SHA256(),
+        length=32,
+        iterations=100000,
+        salt=salt
+    )
+
+    key = base64.urlsafe_b64encode(kdf.derive(bytes(password.encode())))
+    crypt = Fernet(key)
+    
+    # Decrypts the wallet's private key
+    _seed = crypt.decrypt(_seed)
+    
+    # Initialize XRPL wallet using decrypted private key
+    _wallet = wallet.Wallet(seed=_seed.decode(), sequence=0)
```

Comment:
This sequence number seems like it could easily fall out of date (It works for the first transaction, but next time we run this code how do we update the sequence?)

Reply:
(Or is the thought that since we're regenerating wallets it doesn't matter?)

Reply:
`_wallet.sequence = _wallet_seq`: The user has to input in the account's sequence manually to sign a transaction.

Reply:
Why is that done on a separate line? (I wouldn't expect to look elsewhere for the initialization since the `sequence` param is named explicitly)

Reply:
Changes will be made.